var/spack/repos/builtin/packages/jdk/package.py

```diff
@@ -68,3 +68,9 @@ def do_fetch(self, mirror_only=False):
 
     def install(self, spec, prefix):
         distutils.dir_util.copy_tree(".", prefix)
+
+    def setup_environment(self, spack_env, run_env):
+        run_env.set('JAVA_HOME', self.spec.prefix)
```

Comment:
@tgamblin @alfredo-gimenez I think I missed this PR. `run_env` should be moved to `setup_dependent_environment` if you want to have it in every module that depends on `Java`.